File: frontend/src/types/triage.ts
/**
 * Types and constants for error triage operations.
 *
 * - TriageScore: per-sample composite error score from worst-images endpoint
 * - WorstImagesResponse: ranked list wrapper
 * - TRIAGE_OPTIONS: UI-facing tag definitions with color classes
 * - TriageTag: union type of valid triage tag strings
 */

export interface TriageScore {
  sample_id: string;
  error_count: number;
  confidence_spread: number;
  score: number;
}

export interface WorstImagesResponse {
  items: TriageScore[];
}

export const TRIAGE_OPTIONS = [
  { tag: "triage:tp", label: "TP", colorClass: "bg-green-500 hover:bg-green-600", textClass: "text-green-600 dark:text-green-400" },
  { tag: "triage:fp", label: "FP", colorClass: "bg-red-500 hover:bg-red-600", textClass: "text-red-600 dark:text-red-400" },
  { tag: "triage:fn", label: "FN", colorClass: "bg-orange-500 hover:bg-orange-600", textClass: "text-orange-600 dark:text-orange-400" },
  { tag: "triage:mistake", label: "Mistake", colorClass: "bg-amber-500 hover:bg-amber-600", textClass: "text-amber-600 dark:text-amber-400" },
] as const;

export type TriageTag = (typeof TRIAGE_OPTIONS)[number]["tag"];

File: app/models/triage.py
"""Pydantic models for triage operations.

- SetTriageTagRequest: body for PATCH /samples/set-triage-tag
- TriageScore: per-sample composite error score
- WorstImagesResponse: ranked list of worst images
"""

from pydantic import BaseModel

# Triage tag constants
TRIAGE_PREFIX = "triage:"
VALID_TRIAGE_TAGS = {"triage:fp", "triage:tp", "triage:fn", "triage:mistake"}


class SetTriageTagRequest(BaseModel):
    """Request body for setting a triage tag on a single sample."""

    dataset_id: str
    sample_id: str
    tag: str


class TriageScore(BaseModel):
    """Per-sample composite error score for worst-image ranking."""

    sample_id: str
    error_count: int
    confidence_spread: float
    score: float


class WorstImagesResponse(BaseModel):
    """Ranked list of samples by composite error score."""

    items: list[TriageScore]
